### DISCUSSION TITLE
How to silence API URL outputs?

### DISCUSSION THREAD

**Discussion by ezloj** [python-gitlab/python-gitlab#2906] [General]
Logging level does not do it. Gitlab lib seems to be printing API URL for each call it makes. How to prevent that? Tried reading the docs and looking at options and setting logging level - nothing helps

**Comment by nejch** [comment]
@ezloj unless you explicitly call `enable_debug()`, python-gitlab does not log API URLs. 

Are you sure you're using python-gitlab? What's the code snippet you use?

**Comment by ezloj** [comment]
@nejch here is the snippet and I'm 99% sure I'm using `python-gitlab`:

```
GITLAB = gitlab.Gitlab(GITLAB_URL, private_token=os.getenv("GITLAB_TOKEN"))

proj = GITLAB.projects.list(search="bla")
proj = GITLAB.projects.list(search="bla")
proj = GITLAB.projects.list(search="bla")
print(gitlab.__version__)
```

Snippet produces this output:

```
 13:39 > test.py
https://MY_GITLAB_URL_OBFUSCATED/api/v4/projects
https://MY_GITLAB_URL_OBFUSCATED/api/v4/projects
https://MY_GITLAB_URL_OBFUSCATED/api/v4/projects
4.6.0
```

**Reply by max-wittig** [reply]
I've never seen that. Could you test this in an isolated environment? Was some variable set that `request` uses?

**Reply by ezloj** [reply]
We just tested it with a colleague and confirmed it's something on my side. I tried empty new shell with all aliases/variables removed and it still behaves that way 🤷  thank you for the help!

**Reply by nejch** [reply]
You might have been playing with the client code and added some prints in your installed python-gitlab code :)

**Reply by ezloj** [reply]
That was it. I swear I forgot that. Reinstall worked. Thanks a lot 🙏
